File: HireHubDemo/hirehub-core/src/main/java/com/hirehub/core/entities/Job.java
package com.hirehub.core.entities;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "jobs")
@Data
public class Job {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "required_skills", columnDefinition = "TEXT")
    private String requiredSkills;

    @Enumerated(EnumType.STRING)
    private Status status = Status.OPEN;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "company_id", nullable = false)
    private Company company;

    public enum Status {
        OPEN, CLOSED
    }
}

File: HireHubDemo/HireHubDemo/hirehub-referral/Models/Job.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HireHub.Referral.Models
{
    [Table("jobs")]
    public class Job
    {
        [Key]
        public long Id { get; set; }

        [Column("company_id")]
        public long CompanyId { get; set; }

        public string Title { get; set; }
        public string Description { get; set; }
        public string Status { get; set; } // OPEN, CLOSED
    }
}
